### ISSUE TITLE
[Support] Ubuntu client has no access to VPN LAN

### ISSUE THREAD

**Issue by tmh7296** [pivpn/pivpn#1016]
## In raising this issue, I confirm the following: 

`{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting can be *is* directly related to the pivpn installer script.
- [] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Issue
It's _possible_ this issue isn't the fault of PiVpn, but I'm not sure where else to look. A while ago I configured PiVpn and have been using it for the past few months. I generated a config for my android phone using `pivpn add`. Recently I tried to generate another config for my Ubuntu laptop. The laptop appears to successfully connect. I'm able to access the internet while connected, however I have no access to devices on the LAN. Because my android phone is able to access LAN devices, I'm led to believe that this is an issue with my client/the newly generated client config.

### Have you searched for similar issues and solutions?
    (yes/no / which issues?)
Yes. Very few related results.

### Console output of      `curl -L install.pivpn.dev | bash`
```
  Output Here
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
  Output Here
```

### Console output of      `pivpn debug`
```
::: Generating Debug Output
::::		PiVPN debug		 ::::
=============================================
::::		Latest commit		 ::::
commit d0c10db6ec391961b7201fb564055c1176ca73e3
Author: 4s3ti <cfcolaco@colacoweb.net>
Date:   Tue Sep 3 10:09:48 2019 +0200

    install.sh: apt-get with , uninstall.sh: added var PKG_MANAGER and replaced apt-get with
=============================================
::::	    Installation settings    	 ::::
/etc/pivpn/DET_PLATFORM -> Raspbian
/etc/pivpn/FORWARD_CHAIN_EDITED -> 1
/etc/pivpn/HELP_SHOWN -> 
/etc/pivpn/INPUT_CHAIN_EDITED -> 1
/etc/pivpn/INSTALL_PORT -> <PORT>
/etc/pivpn/INSTALL_PROTO -> udp
/etc/pivpn/INSTALL_USER -> pi
/etc/pivpn/NO_UFW -> 1
/etc/pivpn/pivpnINTERFACE -> wlan0
/etc/pivpn/TWO_POINT_FOUR -> 
=============================================
::::	setupVars file shown below	 ::::
INSTALL_USER=pi
UNATTUPG=unattended-upgrades
pivpnInterface=wlan0
IPv4dns=<DNS>
IPv4addr=<Local IP>
IPv4gw=<Local IP>
pivpnProto=udp
PORT=<PORT> ::: Generating Debug Output
::::		PiVPN debug		 ::::
=============================================
::::		Latest commit		 ::::
commit d0c10db6ec391961b7201fb564055c1176ca73e3
Author: 4s3ti <cfcolaco@colacoweb.net>
Date:   Tue Sep 3 10:09:48 2019 +0200

    install.sh: apt-get with , uninstall.sh: added var PKG_MANAGER and replaced apt-get with
=============================================
::::	    Installation settings    	 ::::
/etc/pivpn/DET_PLATFORM -> Raspbian
/etc/pivpn/FORWARD_CHAIN_EDITED -> 1
/etc/pivpn/HELP_SHOWN -> 
/etc/pivpn/INPUT_CHAIN_EDITED -> 1
/etc/pivpn/INSTALL_PORT -> <PORT>
/etc/pivpn/INSTALL_PROTO -> udp
/etc/pivpn/INSTALL_USER -> pi
/etc/pivpn/NO_UFW -> 1
/etc/pivpn/pivpnINTERFACE -> wlan0
/etc/pivpn/TWO_POINT_FOUR -> 
=============================================
::::	setupVars file shown below	 ::::
INSTALL_USER=pi
UNATTUPG=unattended-upgrades
pivpnInterface=wlan0
IPv4dns=<DNS>
IPv4addr=<Local IP>
IPv4gw=<Local IP>
pivpnProto=udp
PORT=<PORT>
ENCRYPT=256
APPLY_TWO_POINT_FOUR=true
DOWNLOAD_DH_PARAM=false
PUBLICDNS=
OVPNDNS1=8.8.8.8
OVPNDNS2=8.8.4.4
=============================================
::::  Server configuration shown below   ::::
dev tun
proto udp
port <PORT>
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/octopi_05bde668-a92b-456a-91f7-81aeb7392c15.crt
key /etc/openvpn/easy-rsa/pki/private/octopi_05bde668-a92b-456a-91f7-81aeb7392c15.key
dh none
topology subnet
server 10.8.0.0 255.255.255.0
# Set your primary domain name server address for clients
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"
# Prevent DNS leaks on Windows
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
keepalive 1800 3600
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
cipher AES-256-CBC
auth SHA256
user nobody
group nogroup
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device. 
#duplicate-cn
# Generated for use by PiVPN.io
=============================================
::::  Client template file shown below   ::::
client
dev tun
proto udp
remote REMOTE <PORT>
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
tls-version-min 1.2
verify-x509-name octopi_05bde668-a92b-456a-91f7-81aeb7392c15 name
cipher AES-256-CBC
auth SHA256
auth-nocache
verb 3
=============================================
:::: 	Recursive list of files in	 ::::
::: /etc/openvpn/easy-rsa/pki shows below :::
/etc/openvpn/easy-rsa/pki/:
ca.crt
crl.pem
Default.txt
ecparams
extensions.temp
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
mobile.ovpn
openssl-easyrsa.cnf
private
renewed
revoked
safessl-easyrsa.cnf
serial
serial.old
ta.key
thinkpad.ovpn

/etc/openvpn/easy-rsa/pki/ecparams:
prime256v1.pem

/etc/openvpn/easy-rsa/pki/issued:
mobile.crt
octopi_05bde668-a92b-456a-91f7-81aeb7392c15.crt
thinkpad.crt

/etc/openvpn/easy-rsa/pki/private:
ca.key
mobile.key
octopi_05bde668-a92b-456a-91f7-81aeb7392c15.key
thinkpad.key

/etc/openvpn/easy-rsa/pki/renewed:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/renewed/private_by_serial:

/etc/openvpn/easy-rsa/pki/renewed/reqs_by_serial:

/etc/openvpn/easy-rsa/pki/revoked:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/revoked/private_by_serial:

/etc/openvpn/easy-rsa/pki/revoked/reqs_by_serial:
=============================================
::::		Self check		 ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] Iptables INPUT rule set
:: [OK] Iptables FORWARD rule set
:: [OK] OpenVPN is running
:: [OK] OpenVPN is enabled (it will automatically start on reboot)
:: [OK] OpenVPN is listening on port <PORT>/udp
=============================================
::::      Snippet of the server log      ::::
=============================================
::::		Debug complete		 ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.txt
::: 
ENCRYPT=256
APPLY_TWO_POINT_FOUR=true
DOWNLOAD_DH_PARAM=false
PUBLICDNS=
OVPNDNS1=8.8.8.8
OVPNDNS2=8.8.4.4
=============================================
::::  Server configuration shown below   ::::
dev tun
proto udp
port <PORT>
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/octopi_05bde668-a92b-456a-91f7-81aeb7392c15.crt
key /etc/openvpn/easy-rsa/pki/private/octopi_05bde668-a92b-456a-91f7-81aeb7392c15.key
dh none
topology subnet
server 10.8.0.0 255.255.255.0
# Set your primary domain name server address for clients
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"
# Prevent DNS leaks on Windows
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
keepalive 1800 3600
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
cipher AES-256-CBC
auth SHA256
user nobody
group nogroup
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device. 
#duplicate-cn
# Generated for use by PiVPN.io
=============================================
::::  Client template file shown below   ::::
client
dev tun
proto udp
remote REMOTE <PORT>
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
tls-version-min 1.2
verify-x509-name octopi_05bde668-a92b-456a-91f7-81aeb7392c15 name
cipher AES-256-CBC
auth SHA256
auth-nocache
verb 3
=============================================
:::: 	Recursive list of files in	 ::::
::: /etc/openvpn/easy-rsa/pki shows below :::
/etc/openvpn/easy-rsa/pki/:
ca.crt
crl.pem
Default.txt
ecparams
extensions.temp
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
mobile.ovpn
openssl-easyrsa.cnf
private
renewed
revoked
safessl-easyrsa.cnf
serial
serial.old
ta.key
thinkpad.ovpn

/etc/openvpn/easy-rsa/pki/ecparams:
prime256v1.pem

/etc/openvpn/easy-rsa/pki/issued:
mobile.crt
octopi_05bde668-a92b-456a-91f7-81aeb7392c15.crt
thinkpad.crt

/etc/openvpn/easy-rsa/pki/private:
ca.key
mobile.key
octopi_05bde668-a92b-456a-91f7-81aeb7392c15.key
thinkpad.key

/etc/openvpn/easy-rsa/pki/renewed:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/renewed/private_by_serial:

/etc/openvpn/easy-rsa/pki/renewed/reqs_by_serial:

/etc/openvpn/easy-rsa/pki/revoked:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/revoked/private_by_serial:

/etc/openvpn/easy-rsa/pki/revoked/reqs_by_serial:
=============================================
::::		Self check		 ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] Iptables INPUT rule set
:: [OK] Iptables FORWARD rule set
:: [OK] OpenVPN is running
:: [OK] OpenVPN is enabled (it will automatically start on reboot)
:: [OK] OpenVPN is listening on port <PORT>/udp
=============================================
::::      Snippet of the server log      ::::
=============================================
::::		Debug complete		 ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.txt
::: 
```
### Have you taken any steps towards solving your issue?
```
  which?
```

**Comment by szaimen** [comment]
I have kind of the same behaviour but I think I can access devices, it is just _really_ slow: something like 20 KB/s connection speed + lots of timouts although the norml internet speed is much faster. 
It is unfortunately unusable in this state.

**Comment by orazioedoardo** [comment]
Can you try to connect the Ubuntu laptop from an external network (like hotspot from your phone)?

**Comment by tmh7296** [comment]
Closing this issue. This was not a pivpn problem. In case anyone runs into the same issue in the future, the local network I was trying to connect from (a friend's house) and the remote network I was trying to connect to (my house) were using the same IP subnets, 192.168.0.x

To fix the issue I logged into my router and changed the subnet that it assigned IPs from. As a result I had to uninstall pivpn, assign the pi a new static IP within the new subnet I assigned, re-install pivpn with the new static IP, generate new OpenVpn configs for all of my clients, and  fix the port forwarding rule on my router to include the new RPI static IP.

Everything works as expected now, I'm able to hit internal IPs on my remote network from my friend's house. What this still doesn't explain, though, is why my android phone had no problems with the previous configuration. It was connected to my friend's wi-fi network as well. Very curious.
